### PR TITLE
[Debug] Restoring back the state of the Debug component (1st step)

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/issues/32371
| License       | MIT

After a good discussion with @nicolas-grekas, we made the decision to split the current `ErrorCatcher` component into several:
 * `ErrorHandler` it would be the Debug component before these changes https://github.com/symfony/symfony/pull/31065, with everything related to ErrorHandler, Debug, DebugClassLoader classes and change its name.
 * `ErrorDumper` it would be the current ErrorCatcher but with FlattenException + the new error renderer system only.

This is the first step, then we can deprecate everything for the Debug component in favor of the ErrorHandler and ErrorDumper components, **BUT without moving any code !!**, that would give us more freedom to do it correctly in the new components.

NOTE: For this PR I've copy the `Debug` component directory from the revision prior to merged commit  https://github.com/symfony/symfony/pull/31065 in 4.4 branch.

### CODE REVIEW

**Comment by fabpot** [comment]
ErrorRendeder instead of ErrorDumper?

**Comment by yceruto** [comment]
I will also remove from the current `ErrorCatcher` everything unrelated to the future `ErrorDumper` and make it work. Later, as last step, rename both components to `ErrorHandler` and `ErrorDumper` respectively.

Status: Needs Work

**Comment by yceruto** [comment]
Status: Needs Review

(low/high Travis's failures are gone after merge)

**Comment by yceruto** [comment]
> ErrorRenderer instead of ErrorDumper?

That one is nice too. I personally like `ErrorDumper` it feel like `VarDumper` but for errors. Let's rename it for another PR.

File: src/Symfony/Component/Debug/composer.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "minimum-stability": "dev",
     "extra": {
         "branch-alias": {
-            "dev-master": "4.4-dev"
+            "dev-master": "4.3-dev"
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Debug/composer.json]: should be reverted

File: src/Symfony/Component/Debug/composer.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
         "symfony/http-kernel": "<3.4"
     },
     "require-dev": {
-        "symfony/http-kernel": "^3.4|^4.0|^5.0"
+        "symfony/http-kernel": "~3.4|~4.0"
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Debug/composer.json]: to be reverted

File: src/Symfony/Component/Debug/Exception/FlattenException.php [hunk 3]
```diff
@@ -12,21 +12,358 @@
...
+     * @deprecated since Symfony 4.4, use Symfony\Component\ErrorCatcher\Exception::createFromThrowable() instead.
+     */
+    public static function create(\Exception $exception, $statusCode = null, array $headers = [])
+    {
+        @trigger_error(sprintf('The "%s()" method is deprecated since Symfony 4.4, use Symfony\Component\ErrorCatcher\Exception::createFromThrowable() instead.', __METHOD__), E_USER_DEPRECATED);
+
+        return static::createFromThrowable($exception, $statusCode, $headers);
+    }
+
+    public static function createFromThrowable(\Throwable $exception, ?int $statusCode = null, array $headers = []): self
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Debug/Exception/FlattenException.php]: please remove the `?`

**Comment by yceruto** [comment]
Thanks Nicolas! now Travis's low tests pass.

Status: Needs Review

**Comment by nicolas-grekas** [comment]
It would make sense to me to remove error-catcher from the composer.json of the Debug component.
This means using the legacy FlattenException in ExceptionHandler, which is OK.

**Comment by yceruto** [comment]
@nicolas-grekas it's done now, thanks.

**Comment by nicolas-grekas** [comment]
Thank you @yceruto.

**Pull request merged**